### PR TITLE
[GO] Refactored functions to return StatusResult

### LINKED ISSUE #1284: Change error handling in Go to be all StatusResult
## 📖 Description
Refactor the Go FromProto, ToProto and other methods from each endpoint to return StatusResult instead of Go error. This will make the codebase more cohesive as well as give errors passed up through the chain more info for handling.

---

## 🎯 Goal / Outcome
No more error returns, only StatusResult

---

## ✅ Definition of Done (DoD)
- [ ] Feature is implemented and works as expected
- [ ] Edge cases are handled
- [ ] Tests are added or updated (if applicable)
- [ ] Documentation is updated (if needed)
- [ ] Code is reviewed and approved -->

---

## 🛠️ Implementation Notes (Optional)
Wait for the merge of https://github.com/rossvideo/Catena/pull/1262

---

## 🔗 Related Issues / Links

---
## 🚧 Risks / Open Questions (Optional)

---

### PR DESCRIPTION
**Pull request by nelgd**
## 📖 Description

Refactored the Go `FromProto`, `ToProto`, and other conversion/utility methods across the SDK to return `StatusResult` instead of Go `error`. This makes the codebase more cohesive and gives errors passed up through the chain richer info (status code + message) for handling.

---

## ✅ Definition of Done (DoD)
- [x] Feature is implemented and works as expected
- [x] Edge cases are handled
- [x] Tests are added or updated (if applicable)

---

### Core `catena` package (`sdks/go/pkg/catena/`)
- **`value.go`** — Changed `ToProto`, `FromProto`, and `ToCatenaValue` to return `StatusResult` instead of `error`. All internal `fmt.Errorf` calls replaced with `StatusResult{Code: ..., Error: "..."}`.
- **`device.go`** — Changed `ToCatenaDevice`, `toProtoDevice` (unexported), and `CatenaDevice.ToJSON` to return `StatusResult`.
- **`asset.go`** — Changed `ToCatenaAsset`, `dataPayloadToProto` (unexported), `CatenaAsset.ToJSON`, `ParsePayloadEncoding`, and `TranscodeAssetPayload` to return `StatusResult`.

### Error-to-StatusCode mapping convention
- Type mismatches, unsupported types, nil values → `INVALID_ARGUMENT`
- Serialization/marshaling failures → `INTERNAL`
- Invalid payload configurations (both/neither payload+URL) → `INVALID_ARGUMENT`
- Compression decode/encode failures → `INTERNAL`

### Server call sites
- **`sdks/go/pkg/rest/server.go`** — Updated `handleValueEndpoint`, `handleAssetEndpoint`, `handleCommandEndpoint`, and `writeAssetResult` to check `StatusResult.Code` instead of `err != nil`.
- **`sdks/go/pkg/grpc/server.go`** — Updated `SetValue`, `MultiSetValue`, and `ExecuteCommand` to check `StatusResult.Code`.
- **`sdks/go/pkg/internal/base_server.go`** — Updated `BroadcastUpdate` to check `StatusResult.Code`.

### Bug fixes
- Fixed 5 pre-existing logging bugs where `err.Error` (from a `StatusResult` returned by `ReadRequestJSON`) was incorrectly logged instead of `errProto` (the actual conversion error) in:
  - REST `handleValueEndpoint`
  - REST `handleCommandEndpoint`
  - gRPC `SetValue`
  - gRPC `MultiSetValue`
  - gRPC `ExecuteCommand`

### Tests
- **`value_test.go`** — Updated ~50+ assertions from `err != nil` / `err == nil` to `err.Code != OK` / `err.Code == OK`, and format strings from `err` to `err.Error`.
- **`device_test.go`** — Updated ~10 assertions.
- **`asset_test.go`** — Updated ~25 assertions (only for `ToCatenaAsset`, `TranscodeAssetPayload`, `ParsePayloadEncoding`, `ToJSON`; compress/decompress helpers left unchanged).
- **`grpc/server_test.go`** and **`grpc/test_helpers_test.go`** — Updated call sites.
- **`rest/server_test.go`** — No changes needed (all relevant calls already discard the second return value).

### Examples (6 files in `sdks/go/examples/`)
- `getSetValue_REST.go` — Updated `ToCatenaValue`, `ToCatenaDevice` calls.
- `getSetValue_GRPC.go` — Updated `ToProto` call.
- `getDevice_REST.go` — Updated `ToCatenaDevice` call.
- `getDevice_GRPC.go` — Updated `ToCatenaDevice` call.
- `getAsset_REST.go` — Updated `ToCatenaAsset` call.
- `executeCommand_REST.go` — No changes needed (all calls discard the result).

---

## 🔗 Related Issues / Links

Closes issue #1284

### CODE REVIEW

File: sdks/go/examples/getSetValue_REST/getSetValue_REST.go [hunk 1]
```diff
@@ -206,8 +206,13 @@ func main() {
 		slot := slot // capture loop variable
 		srv.RegisterExecuteCommandHandler(slot, func(slot uint16, commandFqoid string, payload any) (catena.CommandResult, catena.StatusResult) {
 			logger.Info("ExecuteCommand", "slot", slot, "command", commandFqoid, "payload", payload)
-			catenaVal, err := catena.ToCatenaValue("Command " + commandFqoid + " executed successfully")
-			if err != nil {
+			response := map[string]any{
```
Comment (ctwarog-ross) [inline sdks/go/examples/getSetValue_REST/getSetValue_REST.go]: What is the purpose of this function? Seems it just returns a success?
Everything else looks good

Comment (nelgd) [reply sdks/go/examples/getSetValue_REST/getSetValue_REST.go]: I guess its not needed since we arent using commands for that example. Ill take it out

**Review by ctwarog-ross — APPROVED** [review]
